src/picongpu/include/particles/ionization/ionization.hpp

```diff
@@ -38,110 +38,58 @@
-
-        typedef typename T_BBox::ValueType BType;
-        typedef typename T_EBox::ValueType EType;
-
-        PMACC_AUTO(particle,ionFrame[localIdx]);
-        
-        floatD_X pos = particle[position_];
-        const int particleCellIdx = particle[localCellIdx_];
-
-        DataSpace<TVec::dim> localCell(DataSpaceOperations<TVec::dim>::template map<TVec > (particleCellIdx));
-
-       
-        EType eField = Field2ParticleInterpolation()
-            (eBox.shift(localCell).toCursor(), pos, NumericalCellType::getEFieldPosition());
-        
-        BType bField =Field2ParticleInterpolation()
-            (bBox.shift(localCell).toCursor(), pos, NumericalCellType::getBFieldPosition());
-        
-        /* define number of bound macro electrons before ionization */
-        float_X prevBoundElectrons = particle[boundElectrons_];
- 
-        /* this is the point where actual ionization takes place */
-        IonizeAlgo ionizeAlgo;
-        ionizeAlgo(
-             bField, eField,
-             particle
-             );
-        
-        /* determine number of new macro electrons to be created */
-        newMacroElectrons = prevBoundElectrons - particle[boundElectrons_];
-        /*calculate one dimensional cell index*/
-
-    }
-}; // struct IonizeParticlesPerFrame
 
+using namespace PMacc;
 
 /** kernelIonizeParticles
  * \brief main kernel for ionization
  *
  * - maps the frame dimensions and gathers the particle boxes
- * - caches the E- and B-fields
  * - contains / calls the ionization algorithm
  * - calls the electron creation functors
  *
- * \tparam BlockDescription_ container for information about block / super cell dimensions
  * \tparam ParBoxIons container of the ions
  * \tparam ParBoxElectrons container of the electrons
- * \tparam BBox b-field box class
- * \tparam EBox e-field box class
  * \tparam Mapping class containing methods for acquiring info from the block
- * \tparam FrameIonizer \see IonizeParticlesPerFrame
+ * \tparam FrameIonizer \see e.g. BSI_Impl in BSI_Impl.hpp
+ *         instance of the ionization model functor
  */
-template<class BlockDescription_, class ParBoxIons, class ParBoxElectrons, class BBox, class EBox, class Mapping, class FrameIonizer>
+template<class ParBoxIons, class ParBoxElectrons, class Mapping, class FrameIonizer>
 __global__ void kernelIonizeParticles(ParBoxIons ionBox,
                                       ParBoxElectrons electronBox,
-                                      EBox fieldE,
-                                      BBox fieldB,
                                       FrameIonizer frameIonizer,
                                       Mapping mapper)
 {
-    
+
     /* "particle box" : container/iterator where the particles live in
-     * and where one can get the frame in a super cell from */
+     * and where one can get the frame in a super cell from
+     */
     typedef typename ParBoxElectrons::FrameType ELECTRONFRAME;
```

Comment:
This typedef is not following our naming conventions and are **unused**.

Please remove both lines